miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -47,6 +61,9 @@ internal class RealMiniApp(
         appId.isBlank() -> throw sdkExceptionForInvalidArguments()
         else -> {
             val (basePath, miniAppInfo) = miniAppDownloader.getMiniApp(appId)
+            val versionId = fetchInfo(appId).version.versionId
+            checkWithManifest(appId, versionId)
```

Comment:
There's still some duplicate code in the `create` functions. I think you didn't move all of it because of the `versionId` above? See my comment above above `versionId`, and then I think you can share almost all of the code in a single function.